nautobot/virtualization/views.py

```diff
@@ -242,18 +242,21 @@ def post(self, request, *args, **kwargs):
 #
 
 
-class VirtualMachineListView(generic.ObjectListView):
-    queryset = VirtualMachine.objects.all()
-    filterset = filters.VirtualMachineFilterSet
-    filterset_form = forms.VirtualMachineFilterForm
-    table = tables.VirtualMachineDetailTable
-    template_name = "virtualization/virtualmachine_list.html"
-
-
-class VirtualMachineView(generic.ObjectView):
-    queryset = VirtualMachine.objects.select_related("software_version", "tenant__tenant_group")
+class VirtualMachineUIViewSet(NautobotUIViewSet):
+    bulk_update_form_class = forms.VirtualMachineBulkEditForm
+    filterset_class = filters.VirtualMachineFilterSet
+    filterset_form_class = forms.VirtualMachineFilterForm
+    form_class = forms.VirtualMachineForm
+    serializer_class = serializers.VirtualMachineSerializer
+    table_class = tables.VirtualMachineDetailTable
+    queryset = VirtualMachine.objects.select_related(
+        "cluster", "role", "status", "tenant", "software_version", "tenant__tenant_group"
+    )
 
     def get_extra_context(self, request, instance):
+        if not instance:
```

Comment:
For core, I see we do this in device, so could make sense to do here as well. That being said, could make sense to standardize on the `if self.action == "foobar":`.